### ISSUE TITLE
libmsquic fails to install on Ubuntu 16.04

### ISSUE THREAD

**Issue by premun** [dotnet/arcade#13430]
- [x] This issue is blocking 
- [ ] This issue is causing unreasonable pain

This is an issue that was caused by https://github.com/dotnet/arcade/issues/13130 and manifests as the inability to install OpenSSL 1.1 on some Ubuntu versions. This is needed for the `libmsquic` library which depends on `libssl1.1`.

### Release Note Category
- [ ] Feature changes/additions 
- [x] Bug fixes
- [ ] Internal Infrastructure Improvements
### Release Note Description
Fixed libmsquic installation in Ubuntu 16.04 images

**Comment by premun** [comment]
Usage of the queue is with docker only:
- https://msit.powerbi.com/groups/de8c4cb8-b06d-4af8-8609-3182bb4bdc7c/reports/8b49a89c-7aea-4fdb-95d5-547a9fe5d734/ReportSection70c8fcce141066279901?ctid=72f988bf-86f1-41af-91ab-2d7cd011db47&pbi_source=shareVisual&visual=96716bb8263b19c4e788&height=561.63&width=962.79&bookmarkGuid=13a4bbce-4c7e-4b46-9b11-bc195679720a
- Jobs in the last 30 days:
    ```SQL
    Jobs
    | where QueueName contains "ubuntu.1604.amd64.open" and Started > now()-30d and Properties !contains "@mcr.microsoft.com"
    | order by Finished desc 
    | limit 100
    ```

**Comment by dougbu** [comment]
Isn't there now a `libmsquic` version w/ no `libssl1.1` dependency❔
And, before you moved the `libssl1.1` installation earlier in image builds, weren't we already installing that package -- obviously for other reasons because `libmsquic` was installed first❔

**Comment by premun** [comment]
@dougbu I quite don't understand how we got to this point
- I added the manual installation of `libssl1.1` **only to the 22.04 queue**.
- I bumped the `libssl1.1` library a little but I tested the older version in my PR and it still wasn't installing.
- It's very much possible the 16.04 queue didn't get re-build for quite some time but we changed the `linux-pip` artifact recently which triggered a rebuild of all Linux images and this started failing

Is it maybe possible that meanwhile, while we didn't rebuild 16.04, `libmsquic` in the Microsoft package repository got bumped and started depending on `libssl` but we didn't notice? And now it has this dependency but `libssl` was never in 16.04?

**Comment by premun** [comment]
Regardless, we have noticed that this package is not used on 16.04 as this queue is only used together with docker and only in some servicing branches. So we decided with Ilya and Tomas that we could even remove it if we don't find a way to install it.

However, with the new version of msquic that has no dependency, we could try rebuilding the 16.04 image without this additional libssl and see if it works.
If not, we should remove libmsquic from the 16.04 image.

cc @ali-turan7 @MilenaHristova - this will be Ali's FR task.

**Comment by dougbu** [comment]
what is the current status here❔

**Comment by ali-turan7** [comment]
I took MSQuicArtifacts out of the UbuntuArtifacts template and checked the desired outcome using the OSOB CLI. However, we made the decision to use a custom template for 16.04 without the mentioned package rather than altering the UbuntuArtifacts, which is being used by other versions. This results in reordering the packages and rebuilding all the images; we just want to modify the 16.04 images. I will be working on this problem. 
![MicrosoftTeams-image (1)](https://github.com/dotnet/arcade/assets/128047116/614671bc-329e-4959-8638-34cce5d128b2)
![MicrosoftTeams-image](https://github.com/dotnet/arcade/assets/128047116/711200b6-6bfc-4141-8d21-ca4cca2db7f3)
